### ISSUE TITLE
Retain custom flags during merge

### ISSUE THREAD

**Issue by johantell** [elixir-gettext/gettext#369]
**Background:**
In our project, we've implemented an auto-translate feature that helps team members translate to languages they don't know (and we'll help adjust them through the review process or send them over to someone who helps us with the translation).

We've set up a mix task that back-fills all empty translations in a specific language with translations from an external language model and adds the `auto-translated` tag to the message.

We've noticed recently that all those tags get removed when doing a `mix gettext.extract --merge`, which causes some disruption for us as we use those tags to know which messages we should send to our external human translators for verification.

**Problem**
From my understanding (by looking through the `Merger` module), `fuzzy` is the only flag allowed to be retained over multiple merges (https://github.com/elixir-gettext/gettext/blob/main/lib/gettext/merger.ex#L230-L236), which is why we're experiencing our problem

**Suggested solution**
I understand that there is most likely a very sound reason to resetting the tags on each extraction so my suggestion is that we make the retained titles configurable through our mix-file:

```elixir
def project do
  [
    #...
    gettext: [
      retain_flags: ["fuzzy", "auto-translated"]
    ]
    #...
  ]
end
```

Where we either use `["fuzzy"]` as a default or merge the "internal" flags with the configured value

---

If this suggestion sounds good to you, I'd be happy to implement it and shoot a PR your way :)

**Comment by whatyouhide** [comment]
@johantell yes, the `retain_flags` option sounds great. Let's have `fuzzy` in there by default and keep it in there though, so that you don't have to configure it manually. In that case, we can call the option `:custom_flags_to_keep`, wdyt?

**Comment by johantell** [comment]
Sure thing, `:custom_flags_to_keep` sounds good 👍

**Comment by whatyouhide** [comment]
@johantell awesome. PR is definitely appreciated!! 💟
